cmd/libsnap-confine-private/cgroup-support-test.c

```diff
@@ -0,0 +1,284 @@
+    g_debug("cleanup command: %s", cmd);
+    g_spawn_command_line_sync(cmd, NULL, NULL, NULL, &err);
+    g_assert_no_error(err);
+    g_free(cmd);
+    g_free(fixture->root);
+}
+
+static void test_sc_cgroupv2_is_tracking_happy(cgroupv2_is_tracking_fixture *fixture, gconstpointer user_data) {
+    GError *err = NULL;
+    g_file_set_contents(fixture->p, "0::/foo/bar/baz/snap.foo.app.1234-1234.scope", -1, &err);
+    g_assert_no_error(err);
+
+    /* there exist 2 groups with processes from a given snap */
+    const char *dirs[] = {
+        "/foo/bar/baz/snap.foo.app.1234-1234.scope",
+        "/foo/bar/snap.foo.app.1111-1111.scope",
+        "/foo/bar/bad",
+        "/system.slice/some/app/other",
+        "/user/slice/other/app",
+    };
+
+    for (size_t i = 0; i < sizeof dirs / sizeof dirs[0]; i++) {
+        g_autofree const char *np = g_build_filename(fixture->root, dirs[i], NULL);
+        int ret = g_mkdir_with_parents(np, 0755);
+        g_assert_cmpint(ret, ==, 0);
+    }
+
+    bool is_tracking = sc_cgroup_v2_is_tracking_snap("foo");
+    g_assert_true(is_tracking);
+}
+
+static void test_sc_cgroupv2_is_tracking_just_own_group(cgroupv2_is_tracking_fixture *fixture,
+                                                        gconstpointer user_data) {
+    GError *err = NULL;
+    g_file_set_contents(fixture->p, "0::/foo/bar/baz/snap.foo.app.1234-1234.scope", -1, &err);
+    g_assert_no_error(err);
+
+    /* our group is the only one for this snap */
+    const char *dirs[] = {
+        "/foo/bar/baz/snap.foo.app.1234-1234.scope",
+        "/foo/bar/bad",
+        "/system.slice/some/app/other",
+        "/user/slice/other/app",
+    };
+
+    for (size_t i = 0; i < sizeof dirs / sizeof dirs[0]; i++) {
+        g_autofree const char *np = g_build_filename(fixture->root, dirs[i], NULL);
+        int ret = g_mkdir_with_parents(np, 0755);
+        g_assert_cmpint(ret, ==, 0);
+    }
+
+    bool is_tracking = sc_cgroup_v2_is_tracking_snap("foo");
+    /* our own group is skipped */
+    g_assert_false(is_tracking);
+}
+
+static void test_sc_cgroupv2_is_tracking_no_dirs(cgroupv2_is_tracking_fixture *fixture, gconstpointer user_data) {
+    GError *err = NULL;
+    g_file_set_contents(fixture->p, "0::/foo/bar/baz/snap.foo.app.scope", -1, &err);
+    g_assert_no_error(err);
+
+    bool is_tracking = sc_cgroup_v2_is_tracking_snap("foo");
+    g_assert_false(is_tracking);
+}
+
+static void test_sc_cgroupv2_is_tracking_bad_self_group(cgroupv2_is_tracking_fixture *fixture,
+                                                        gconstpointer user_data) {
+    GError *err = NULL;
+    /* trigger a failure in own group handling */
+    g_file_set_contents(fixture->p, "", -1, &err);
+    g_assert_no_error(err);
+
+    if (g_test_subprocess()) {
+        sc_cgroup_v2_is_tracking_snap("foo");
+    }
+    g_test_trap_subprocess(NULL, 0, 0);
+    g_test_trap_assert_failed();
+    g_test_trap_assert_stderr("cannot obtain own group path\n");
```

Comment:
(nitpick) Maybe we could make this error provide a little bit more context? Like `cannot obtain own group path while looking for cgroup v2 scope/group` or something like this?